tools/shared/src/main/scala/org/scalajs/core/tools/linker/Linker.scala

```diff
@@ -73,4 +73,77 @@ final class Linker(frontend: LinkerFrontend, backend: LinkerBackend)
   }
 }
 
-object Linker extends LinkerPlatformExtensions
+object Linker extends LinkerPlatformExtensions {
+  /** Configuration to be passed to the `apply()` method. */
+  final class Config private (
+      /** Whether to generate source maps. */
+      val sourceMap: Boolean,
+      /** Whether to use the Scala.js optimizer. */
+      val optimizer: Boolean,
+      /** Whether things that can be parallelized should be parallelized.
+       *  On the JavaScript platform, this does not have any effect.
+       */
+      val parallel: Boolean,
+      /** Whether to use the Google Closure Compiler pass.
+       *  On the JavaScript platform, this does not have any effect.
```

Comment:
What about this:

1. Rename `closureCompiler` (resp. `withClosureCompiler`) to `closureCompilerIfAvailable` (resp. `withClosureCompilerIfAvailable`).
2. Add an implicit pimp only in `jvm/`:

```scala
implicit class ConfigPlatformOps(val self: Config) {
  def closureCompiler: Boolean =
    self.closureCompilerIfAvailable
  def withClosureCompiler(closureCompiler: Boolean): Config =
    self.withClosureCompilerIfAvailable(closureCompiler)
}
```
and possibly a read-only pimp in `js/`:
```scala
implicit class ConfigPlatformOps(val self: Config) {
  def closureCompiler: Boolean = false
}
```
This should give all of:
1. No code duplication
2. Default, short name is fully typechecked
3. There is longer, very explicit name for the ability to write cross-platform code.